### PR TITLE
Update webpack, webpack-cli, chokidar, chokidar-cli, and helmet to latest version

### PR DESCRIPTION
**Pull request by mmmavis**
@Pomax I finally got this working. If this PR becomes too hard to review, let me know and I can quickly walk you through it.

---

Related issue: #1162 and #1159

- Updated `webpack`, `webpack-cli`, `chokidar`, `chokidar-cli`, and `helmet` to latest version.
- Added `prettier` as `devDependencies` as now we have to explicitly set it.
- Split webpack config file into "production" and "dev" versions. See this [comment](https://github.com/mozilla/network-pulse/pull/1165/files#r351077205) for details.
- Updated npm tasks so we can run proper tasks depending on the production/dev environment.

---

- `npm test` passed
- watch tasks are still working
- ~~app runs normally~~ currently not working on staging ⚠️

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,20 +1,7 @@
...
-      "process.env": {
-        NODE_ENV: JSON.stringify(`production`)
-      }
-    }),
-    new webpack.optimize.UglifyJsPlugin()
-  ];
-}
-
 module.exports = {
+  mode: process.env.NODE_ENV,
```
Comment (mmmavis) [inline webpack.config.js]: https://webpack.js.org/migrate/4/#mode

File: webpack.server.config.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ var fs = require(`fs`);
 var path = require(`path`);
 
 module.exports = {
+  mode: process.env.NODE_ENV,
```
Comment (mmmavis) [inline webpack.server.config.js]: https://webpack.js.org/migrate/4/#mode

File: webpack.config.js [hunk 3]
```diff
@@ -1,20 +1,7 @@
 var webpack = require(`webpack`);
 
-var plugins = [];
```
Comment (mmmavis) [inline webpack.config.js]: https://webpack.js.org/migrate/4/#deprecatedremoved-plugins

File: package.json [hunk 4]
```diff
@@ -54,8 +54,8 @@
   },
   "homepage": "https://github.com/mozilla/network-pulse#readme",
   "devDependencies": {
-    "chokidar": "^2.1.5",
-    "chokidar-cli": "^1.2.2",
+    "chokidar": "^3.3.0",
```
Comment (mmmavis) [inline package.json]: I don't see any breaking updates except [this one](https://github.com/paulmillr/chokidar/blob/master/.github/full_changelog.md#chokidar-310-sep-16-2019)
> Breaking: dotfiles are no longer filtered out by default. Use ignored option if needed.

But we don't have any dotfiles in the repo so no action needed on our end.

File: package.json [hunk 5]
```diff
@@ -54,8 +54,8 @@
...
+    "chokidar-cli": "^2.1.0",
```
Comment (mmmavis) [inline package.json]: Didn't see any breaking changes either: https://github.com/kimmobrunfeldt/chokidar-cli/releases

File: package.json [hunk 6]
```diff
@@ -80,7 +82,7 @@
     "event-stream": "3.3.4",
     "express": "^4.16.4",
     "font-awesome": "^4.7.0",
-    "helmet": "^3.18.0",
+    "helmet": "^3.21.2",
```
Comment (mmmavis) [inline package.json]: no breaking changes https://github.com/helmetjs/helmet/blob/master/CHANGELOG.md

File: package.json [hunk 7]
```diff
@@ -26,12 +30,11 @@
...
-    "watch:js": "chokidar \"app.jsx\" \"main.jsx\" \"js/**/*.js\" \"components/**/*.js?(x)\" \"pages/**/*.js?(x)\" \"webpack.config.js\" -c \"npm run build-client:js\"",
+    "watch:js": "chokidar \"app.jsx\" \"main.jsx\" \"js/**/*.js\" \"components/**/*.js?(x)\" \"pages/**/*.js?(x)\" \"webpack.**.js\" -c \"npm run build-js:dev:client\"",
     "test": "run-s test:**",
-    "test:eslint": "eslint --config ./.eslintrc.json  \"*.js\" \"*.jsx\" \"js/**/*.js\" \"components/**/*.jsx\" \"pages/**/*.jsx\" \"pages/**/*.js\" webpack.config.js",
+    "test:eslint": "eslint --config ./.eslintrc.json  \"*.js\" \"*.jsx\" \"js/**/*.js\" \"components/**/*.jsx\" \"pages/**/*.jsx\" \"pages/**/*.js\" webpack.**.js",
     "test:scss:styleguide:color": "stylelint \"components/**/*.scss\" \"pages/**/*.scss\" \"scss/**/*.scss\" \"!scss/foundation-styles/_variables.scss\" \"!scss/**/variables.scss\" \"!scss/overrides/_variables.scss\" --syntax scss --config .stylelintrc-colors.js",
-    "test:scss": "stylelint --config .stylelintrc \"components/**/*.scss\" \"pages/**/*.scss\" \"scss/**/*.scss\" --syntax scss",
-    "postinstall": "npm run build"
```
Comment (mmmavis) [inline package.json]: replaced by `heroku-postbuild` task

File: package.json [hunk 8]
```diff
@@ -5,15 +5,19 @@
   "main": "dist/server.bundle.js",
   "scripts": {
     "bootstrap": "shx rm -rf dist && shx mkdir dist",
-    "start": "npm run build && run-p server watch:**",
+    "start": "npm run build:dev && run-p server watch:**",
     "build": "run-s bootstrap build:**",
-    "heroku-postbuild": "echo Skip builds on Heroku",
+    "heroku-postbuild": "npm run build:prod",
+    "build:prod": "run-s bootstrap build-client:** build-js:prod:**",
```
Comment (mmmavis) [inline package.json]: Webpack v4 introduced a new option `mode` and it has to be explicitly set (e.g., we can utilize variable like `process.env.NODE_ENV`.) 

I used this [guide](https://webpack.js.org/guides/production/#setup) to break our existing webpack config into "prod" and "dev" version.

File: webpack.client.common.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
```
Comment (mmmavis) [inline webpack.client.common.js]: Pretty much the same as [what we used to have](https://github.com/mozilla/network-pulse/blob/master/webpack.config.js). I removed plugin related code that's no longer need based the [migration guide](https://webpack.js.org/migrate/4/#deprecatedremoved-plugins).

File: webpack.client.dev.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+const merge = require("webpack-merge");
+const common = require("./webpack.client.common.js");
+
+module.exports = merge(common, {
+  mode: `development`,
```
Comment (mmmavis) [inline webpack.client.dev.js]: https://webpack.js.org/migrate/4/#mode

File: package.json [hunk 11]
```diff
@@ -5,15 +5,19 @@
   "main": "dist/server.bundle.js",
   "scripts": {
     "bootstrap": "shx rm -rf dist && shx mkdir dist",
-    "start": "npm run build && run-p server watch:**",
+    "start": "npm run build:dev && run-p server watch:**",
     "build": "run-s bootstrap build:**",
-    "heroku-postbuild": "echo Skip builds on Heroku",
+    "heroku-postbuild": "npm run build:prod",
```
Comment (Pomax) [inline package.json]: Is this a safe change? The idea of heroku-postbuild is that it triggers _after_ a build, not for it to trigger a new build =x

File: package.json (hunk 7 above)
Comment (Pomax) [reply package.json]: Let's roll this particular change back, given that staging doesn't work right now. And then we can always optimize this later.

File: package.json (hunk 11 above)
Comment (mmmavis) [reply package.json]: ah I followed what we have for foundation site but will revert this and file changes as a followup instead

File: package.json (hunk 7 above)
Comment (mmmavis) [reply package.json]: kk... reverted... sort of.

I have to make `postinstall` build with `production` version of webpack config for Heroku apps to run in the right mode (e.g., `production`). This change does introduce a side effect -- if you run `npm install` locally, Pulse will get built in `production` mode which is not what we want. I'll file a follow-up ticket so this PR doesn't grow any bigger.

Comment (Pomax) [reply package.json]: good call

**Pull request merged**